### PR TITLE
[Core] Separate `SpatialSearch` into header and source and register methods

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

`SpatialSearch ` is exposed to python, but just constructor, this PR exposes the methods in python. Also separates header/source file.

Includes #10963

**🆕 Changelog**

- [Separate SpatialSearch into header and source](https://github.com/KratosMultiphysics/Kratos/commit/08170e0511ab170be0038a9d1b64674c25a97456)
- Expose methods to python
- #10963

### CODE REVIEW

File: kratos/spatial_containers/spatial_search.h [hunk 1]
```diff
@@ -12,1055 +12,1151 @@
...
-/** Detail class definition.
+/**
+* @class SpatialSearch
+* @ingroup KratosCore
+* @brief This class is used to search for elements, conditions and nodes in a given model part
+* @author Carlos Roig
 */
-
-class SpatialSearch 
+class SpatialSearch
```
Comment (roigcarlo) [inline kratos/spatial_containers/spatial_search.h]:
```suggestion
class KRATOS_API(KRATOS_CORE) SpatialSearch
```

**Review by roigcarlo — COMMENTED** [review]
If you want to make the split you will have to mark all the derived classes function as overrides or it won't compile. Other than that :+1:

File: kratos/spatial_containers/spatial_search.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/spatial_containers/spatial_search.h]: I forgot

**Comment by loumalouomega** [comment]
> If you want to make the split you will have to mark all the derived classes function as overrides or it won't compile. Other than that 👍

Yes I am on it

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**